### PR TITLE
Feature/form control changes 2

### PR DESCRIPTION
**Pull request by aamirhm**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)
Jira ticket reference: https://tools.hmcts.net/jira/browse/EUI-6740
We have added a service dropdown so users can select (add & remove) the required service from the list.

### CODE REVIEW

**Review by DavidJayakumar — APPROVED** [review]

File: projects/exui-common-lib/src/lib/components/find-service/find-service.component.spec.ts [hunk 1]
```diff
@@ -1,28 +1,159 @@
...
+    {
+      key: '03',
+      label: '03-label',
+    },
+    {
+      key: null,
+      label: '02-label',
+    },
+  ]
+    component.services = services;
```
Comment (MateiHMCTS) [inline projects/exui-common-lib/src/lib/components/find-service/find-service.component.spec.ts:49]: I think component.services shouldn't be set directly? it should be set from component.field.options

File: projects/exui-common-lib/src/lib/components/find-service/find-service.component.ts [hunk 2]
```diff
@@ -1,30 +1,97 @@
...
+  @Input() public fields: FilterFieldConfig[];
+  @Input() public serviceTitle = 'Choose a service';
+  @Input() public form: FormGroup;
+  @Input() public services: FilterConfigOption[] = [];
+  @Input() public selectedServices: FilterConfigOption[] = [];
+  @Input() public enableAddServiceButton: boolean = true;
+  @Input() public disabled: boolean = false;
+
+  @Output() public serviceFieldChanged = new EventEmitter<void>();
+  @Output() public onError = new EventEmitter<void>();
```
Comment (MateiHMCTS) [inline projects/exui-common-lib/src/lib/components/find-service/find-service.component.ts:21]: You don't need @Output() error. this is managed in the FormGroup/FilterService

File: projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts [hunk 3]
```diff
@@ -297,8 +298,8 @@ export class GenericFilterComponent implements OnInit, OnDestroy {
       if (field.type === 'checkbox' || field.type === 'checkbox-large') {
         const formArray = this.buildCheckBoxFormArray(field, settings);
         this.form.addControl(field.name, formArray);
-      } else if (field.type === 'find-location') {
-        const formArray = this.buildFindLocationFormArray(field, settings);
+      } else if (field.type === 'find-location' || field.type === 'find-service') {
+        const formArray = this.buildFindLocationAndServiceFormArray(field, settings);
```
Comment (MateiHMCTS) [inline projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts:302]: can you please rename buildFindLocationAndServiceFormArray to something more generic if it fits multiple cases? (i.e. buildFormArray)

File: projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts [hunk 4]
```diff
@@ -414,4 +415,8 @@ export class GenericFilterComponent implements OnInit, OnDestroy {
       this.filterService.givenErrors.next(errors);
     }
   }
+
+  public setError(): void {
```
Comment (MateiHMCTS) [inline projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts:419]: Don't need this, errors are managed in FormGroup and FilterService

File: projects/exui-common-lib/src/lib/components/search-service/search-service.component.spec.ts [hunk 5]
```diff
@@ -1,20 +1,27 @@
...
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [ReactiveFormsModule, RouterTestingModule.withRoutes([]), MatAutocompleteModule, MatOptionModule],
+      imports: [
+        FormsModule,
+        ReactiveFormsModule,
+        RouterTestingModule.withRoutes([]),
+        MatAutocompleteModule,
+        MatOptionModule,
+        BrowserAnimationsModule,
```
Comment (MateiHMCTS) [inline projects/exui-common-lib/src/lib/components/search-service/search-service.component.spec.ts:20]: Unused imports

File: projects/exui-common-lib/src/lib/components/search-service/search-service.component.ts [hunk 6]
```diff
@@ -1,26 +1,17 @@
-import { Component, Input } from '@angular/core';
-import { FormGroup } from '@angular/forms';
+import { Component, EventEmitter, Input, Output } from "@angular/core";
+import { FilterConfigOption } from "../../models";
 
 @Component({
-  selector: 'exui-search-service',
-  templateUrl: './search-service.component.html',
-  styleUrls: ['./search-service.component.scss']
+  selector: "exui-search-service",
```
Comment (MateiHMCTS) [inline projects/exui-common-lib/src/lib/components/search-service/search-service.component.ts:5]: We prefer single quote

**Review by MateiHMCTS — CHANGES_REQUESTED** [review]

File: projects/exui-common-lib/src/lib/components/find-service/find-service.component.spec.ts [hunk 7]
```diff
@@ -1,28 +1,159 @@
...
   let fixture: ComponentFixture<FindServiceComponent>;
+  let service = {
+    key: '01',
+    label: '01-label',
+  };
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [ReactiveFormsModule, MatAutocompleteModule, MatOptionModule],
+      imports: [ReactiveFormsModule, MatAutocompleteModule, MatOptionModule, FormsModule],
```
[the remark this answers is not in the document]
Comment (aamirhm) [reply projects/exui-common-lib/src/lib/components/find-service/find-service.component.spec.ts]: I've removed them.

File: projects/exui-common-lib/src/lib/components/find-service/find-service.component.spec.ts (hunk 1 above)
Comment (aamirhm) [reply projects/exui-common-lib/src/lib/components/find-service/find-service.component.spec.ts:48]: This is spec to setting its mock values etc

File: projects/exui-common-lib/src/lib/components/find-service/find-service.component.ts (hunk 2 above)
Comment (aamirhm) [reply projects/exui-common-lib/src/lib/components/find-service/find-service.component.ts]: As discussed, I have removed it.

File: projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts (hunk 3 above)
Comment (aamirhm) [reply projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts]: Updated as suggested.

File: projects/exui-common-lib/src/lib/components/search-service/search-service.component.spec.ts (hunk 5 above)
Comment (aamirhm) [reply projects/exui-common-lib/src/lib/components/search-service/search-service.component.spec.ts]: Removed.

File: projects/exui-common-lib/src/lib/components/search-service/search-service.component.html [hunk 8]
```diff
@@ -1,15 +1,10 @@
...
-    <mat-option *ngFor="let service of services" (onSelectionChange)="onSelectionChange()">
-      {{ service.name }}
-    </mat-option>
-    <mat-option *ngIf="!services?.length && showAutocomplete && term && term.length >= this.minSearchCharacters">No results found</mat-option>
-  </mat-autocomplete>
-</div>
+<div class="govuk-form-group">
+  <select id="serviceSearch__select" class="govuk-select" name="serviceSearch__select"
+    (change)="onSelectionChanged($event.target.value)">
+    <option [value]="null"></option>
```
[the remark this answers is not in the document]
Comment (aamirhm) [reply projects/exui-common-lib/src/lib/components/search-service/search-service.component.html]: I've changed this to 'All' as it is part of AC in our ticket. 
image

File: projects/exui-common-lib/src/lib/components/search-service/search-service.component.html [hunk 9]
```diff
@@ -1,15 +1,10 @@
...
+    <option *ngFor="let service of services" [value]="service.key">
+      {{ service.label }}
+    </option>
+    <option *ngIf="!services?.length">No results found</option>
```
[the remark this answers is not in the document]
Comment (aamirhm) [reply projects/exui-common-lib/src/lib/components/search-service/search-service.component.html]: I've removed it.

**Review by MateiHMCTS — APPROVED** [review]

**Pull request merged**